src/raw/mod.rs

```diff
@@ -29,6 +35,23 @@ impl<K, V> Table<K, V> {
         self.bins.len()
     }
 
+    pub(crate) fn get_moved<'g>(
+        &'g self,
+        for_table: *const Table<K, V>,
+        guard: &'g Guard,
+    ) -> Shared<'g, BinEntry<K, V>> {
+        match self.moved.load(Ordering::SeqCst, guard) {
+            s if s.is_null() => {
+                // if a BinEntry::Moved associated with this table does not yet exist,
+                // create one and store it in `self.moved`
+                self.moved
+                    .store(Owned::new(BinEntry::Moved(for_table)), Ordering::SeqCst);
+                self.moved.load(Ordering::SeqCst, guard)
```

Comment:
We can probably avoid the extra `load` here. Also, I wonder if the `store` should be a `compare_and_swap` in case this can race?

Reply:
How would you get from `Owned` to `Shared` without this? Both `store` and `compare_and_swap` consume the stored pointer. I'm probably missing something, but I don't see it atm

Reply:
[`Owned::into_shared`](https://docs.rs/crossbeam-epoch/0.8.0/crossbeam_epoch/struct.Owned.html#method.into_shared)